iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2504,4 +2504,203 @@ TYPED_TEST(stringTyped_test, AccessMaxPositionOfNotEmptyStringViaConstSubscriptO
 
     EXPECT_THAT(sut[STRINGCAP - 1], Eq('L'));
 }
+
+/// @note template <typename T>
+/// constexpr bool insert(const uint64_t pos, const T& str, const uint64_t count) noexcept
+TEST(String10, InsertStringLiteralAtTheBeginningOfTheStringSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8f3c5bf8-2367-46c0-b803-570a2a640c93");
+    const string<10> expectedString("Hypnotoad");
+    string<10> sut("toad");
+    ASSERT_TRUE(sut.insert(0, "Hypno", 5));
+    EXPECT_THAT(sut.size(), Eq(expectedString.size()));
+    EXPECT_THAT(sut, Eq(expectedString));
+}
+
+TEST(String10, InsertStringLiteralInTheMiddleOfTheStringSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "861eec68-3e78-42cf-ac75-30c23a3e60d2");
+    const string<10> expectedString("Hypnotoad");
+    string<10> sut("Hypoad");
+    ASSERT_TRUE(sut.insert(3, "not", 3));
+    EXPECT_THAT(sut.size(), Eq(expectedString.size()));
+    EXPECT_THAT(sut, Eq(expectedString));
+}
+
+TEST(String10, InsertStringLiteralAtTheEndOfTheStringSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "91f733e0-5ded-4f85-9727-241b649bd7dc");
+    const string<10> expectedString("Hypnotoad");
+    string<10> sut("Hypno");
+    ASSERT_TRUE(sut.insert(sut.size(), "toad", 4));
+    EXPECT_THAT(sut.size(), Eq(expectedString.size()));
+    EXPECT_THAT(sut, Eq(expectedString));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralToEmptyStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "eba320fc-9909-4043-a944-7a5c9f2a4272");
+    const string<1> expectedString("M");
+    ASSERT_TRUE(this->testSubject.insert(0, "M", 1));
+    EXPECT_THAT(this->testSubject.size(), Eq(expectedString.size()));
+    EXPECT_THAT(this->testSubject, Eq(expectedString));
+}
+
+TYPED_TEST(stringTyped_test, InsertEmptyStringLiteralDoesNotChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "37d723b7-5b35-4f1d-ac33-23dd483121a1");
+    this->testSubject = "M";
```

Comment:
Sometimes we use a member as sut and sometimes we create a local sut. This is somewhat inconsistent but ok for now. We should strive to be consistent in future tests. I propose using a local sut if the setup is simple and a member if it is complex.

Nothing to do here, carry on.